### PR TITLE
Validation branch to remove bias in na_cutoff and quantify effects of special sauce

### PR DESCRIPTION
**Pull request by matthewhirschey**
This PR has 3 components. 

1. I've added validation.Rmd and changed  depmap_explore.Rmd, both in the EDA folder, that is "validation" code to help us identify hits to validate in the lab. Hence the name of this branch. Nothing to review here as these do not change the behavior of the app.

2. I've added a new script called `find_threshold.R`. The idea here is to remove bias in identifying the na_cutoff value, which is hard coded as a variable in `current_release.R`. This code should take the URLs in current_release.R, download the data, and then generate a correlation matrix on the raw data, remove zero-expression values, and then calculates the relationship between number of cells lines the gene gets its correlation value from vs. the number of genes that are above the resampled mean +3SD. Then it removes the top 5%, with the working assumption that these are false-positives (5% FDR is a reasonable statistical approach), which then finally spits out a value of the number of cell lines where this 5% line crosses. 

This is going to change my workflow, slightly. The new quarterly update process will begin with getting the URLs of the new datasets entering them in `current_release.R`, running `find_threshold.R` (takes ~1hr to run; importantly, the na_cutoff value from current_release is not called in find_threshold.R), and then entering the printed value in `current_release.R` for generate_x.R scripts to use. 

Candidly, I'd like to be a little more aggressive in the cutoff, as I think there is still a bunch of garbage in there, but this will be a good start. 

3. I've updated the methods to reflect these changes, and to add new graphs that quantifies some of these changes. 

fixes issue #69

### CODE REVIEW

**Comment by matthewhirschey** [comment]
One note @johnbradley, I've added on package called `patchwork` to the methods document used to place two graphs next to eachother. This is never called in the app, so not sure if you want to add it to the master container build or not. So far, I've been making the methods documents locally, and serving them as markdown and image files.

**Comment by matthewhirschey** [comment]
What would be useful @johnbradley would be to:

- [ ] run `find_threshold.R` to get the new `na_cutoff` value
- [ ] make sure that value is then typed into `current_release.R`
- [ ] If nothing fails, then re-run the `generate_x.R scripts`

**Comment by johnbradley** [comment]
@matthewhirschey I ran `find_threadshold.R` and got `631` which is the value already in `current_release.R`. I'm running the generate_*.R scripts now.

If `find_threadshold.R` is the definitive source for the `na_cutoff` value should `find_threadshold.R` save `na_cutoff` to a file instead of printing? This would avoid the possibility of forgetting or mistyping the na_cutoff value.

File: code/current_release.R [hunk 1]
```diff
@@ -2,4 +2,4 @@ release <- "20Q1"
 achilles_url <- "https://ndownloader.figshare.com/files/21521910" #achilles_gene_effect.csv
 ccle_url <- "https://ndownloader.figshare.com/files/21521940" #CCLE_expression.csv
 cclemeta_url <- "https://ndownloader.figshare.com/files/21522000" #sample_info.csv
-na_cutoff <- 631 #~5% FDR
\ No newline at end of file
+na_cutoff <- readRDS(file = here::here("data", paste0(release, "_na_cutoff.Rds"))) #~5% FDR
```
Comment (johnbradley) [inline code/current_release.R]: I think you have a chicken and egg problem here.  This file is sourced by `find_threadshold.R` which will create the 20Q2_na_cutoff.Rds file but fails with `No such file or directory`.
One possible solution is to only set na_cutoff if the file exists:
```suggestion
na_cutoff_file = here::here("data", paste0(release, "_na_cutoff.Rds"))
if (file.exists(na_cutoff_file)) {
  na_cutoff <- readRDS(file = na_cutoff_file) #~5% FDR  
}
```

**Comment by matthewhirschey** [comment]
That’s a great suggested change. I thought about the same chicken/egg problem, but hadn’t considered the problem that it might fail if no file exists. At first I did not specifically name the file with the current release prefix, so there would always be a file there. But then when I changed it to record the release by appending 20Q1, I see the failure with 20Q2.

On Mar 9, 2020, at 7:56 AM, John Bradley <notifications@github.com> wrote:

﻿

@johnbradley commented on this pull request.

________________________________

In code/current_release.R<https://urldefense.com/v3/__https://github.com/matthewhirschey/ddh/pull/71*discussion_r389614231__;Iw!!OToaGQ!8j6wr4OB-pqkAw_sQtU0CdcTfK18A6H-_Nra8E-wCjs0nkgVmzkVwLN01B5BqPSjy-REhBU$>:

> \ No newline at end of file
+na_cutoff <- readRDS(file = here::here("data", paste0(release, "_na_cutoff.Rds"))) #~5% FDR


I think you have a chicken and egg problem here. This file is sourced by find_threadshold.R which will create the 20Q2_na_cutoff.Rds file but fails with No such file or directory.
One possible solution is to only set na_cutoff if the file exists:

⬇️ Suggested change

-na_cutoff <- readRDS(file = here::here("data", paste0(release, "_na_cutoff.Rds"))) #~5% FDR
+na_cutoff_file = here::here("data", paste0(release, "_na_cutoff.Rds"))
+if (file.exists(na_cutoff_file)) {
+  na_cutoff <- readRDS(file = na_cutoff_file) #~5% FDR
+}

**Review by johnbradley — APPROVED** [review]
LGTM

I regenerated the data and didn't see any issues running app.R locally.

**Pull request merged**